src/main/java/org/wildfly/security/auth/realm/KeyStoreBackedSecurityRealm.java

```diff
@@ -72,7 +79,47 @@ public KeyStoreBackedSecurityRealm(final KeyStore keyStore, final Supplier<Provi
 
     @Override
     public RealmIdentity getRealmIdentity(final Principal principal) throws RealmUnavailableException {
-        return principal instanceof NamePrincipal ? new KeyStoreRealmIdentity(principal.getName()) : RealmIdentity.NON_EXISTENT;
+        if (principal instanceof NamePrincipal) {
+            return new KeyStoreRealmIdentity(principal.getName());
+        } else {
+            final X500Principal x500Principal = X500PrincipalUtil.asX500Principal(principal);
+            if (x500Principal == null) {
+                return RealmIdentity.NON_EXISTENT;
+            } else {
+                final KeyStore keyStore = this.keyStore;
+                try {
+                    final Enumeration<String> aliases = keyStore.aliases();
+                    while (aliases.hasMoreElements()) {
```

Comment:
Why is not the check of presence of certificate with given alias left for the next processing, like in case `(principal instanceof NamePrincipal) == true`?